### ISSUE TITLE
 Consider adding Spring Config Server integration

### ISSUE THREAD

**Issue by Walliee** [promregator/promregator#103]
## Summary / Problem Statement
Promregator is published as an executable artifact (uberJAR/docker image), it would be really useful if there was support for sourcing configuration from a centralized repository. 

Since Promregator is a Spring Boot application, a natural choice for this is to use `spring-cloud-config-client`. 

The implementation would just entail including the [spring-cloud-config-starter](https://github.com/spring-cloud/spring-cloud-config#client-side-usage) in the pom.xml. 
Then at startup, Promregator would try to look for a configs at `spring.cloud.config.uri` or `localhost:8080` but continue if none of those resources serve any config. 

Spring Cloud Config can even be [disabled](https://github.com/spring-cloud/spring-cloud-config/blob/6b99631f78bac4e1b521d2cc126c8b2da45d1f1f/spring-cloud-config-client/src/main/java/org/springframework/cloud/config/client/ConfigClientProperties.java#L62) for those who are not interested in this functionality.

**Comment by martin-schaefer** [comment]
I would welcome the support of Spring Config Server as well.

**Comment by eaglerainbow** [comment]
Adding config server support most likely isn't that complicated:

* spring-cloud-starter is already included, since we have `{cipher}` support in the meantime (on `master`)
* Only the `promregator.sh` might require some tweaking 
* Documentation simply is missing.
* The challenging part may be user authentication between Promregator (config client) and the Config Server (I have seen that Cloud Config has quite some support for that, but authentication is not enforced, and I have quite some mental problems with transferring a non-encrypted password for the CF via a potentially unencrypted and unauthenticated channel). 

However, the big headache generator is test coverage: Releasing such a feature without test coverage is bad. Moreover, I lack an idea of how we could do such a test in the unit test environment. This is a classical case for an integration test, for which we are currently missing even a concept.

**Comment by martin-schaefer** [comment]
Yes, but as mentioned by Walliee the spring cloud config client can be disabled. I think it is an 
acceptable risk to integrate it, also because it is tested by spring.io anyway.

**Comment by eaglerainbow** [comment]
> I think it is an acceptable risk to integrate it, also because it is tested by spring.io anyway.

In general, I agree. 
The problem is only that I am currently being bitten by exactly these kind of regression cases: You think that it still runs end-to-end, because all modules/pieces are tested individually - but unfortunately, ... well you know 😁
